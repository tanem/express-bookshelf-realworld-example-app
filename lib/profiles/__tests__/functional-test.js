'use strict';

const chance = require('chance').Chance('profile');
const request = require('supertest');
const app = require('../../app');

const {users} = app.locals.services;

describe('profiles', () => {
  let userOne;
  let userTwo;

  beforeEach(async () => {
    userOne = await users.create({
      bio: chance.profession(),
      email: chance.email(),
      image: chance.avatar(),
      password: chance.word({length: 8}),
      username: chance.word(),
    });
    userTwo = await users.create({
      bio: chance.profession(),
      email: chance.email(),
      image: chance.avatar(),
      password: chance.word({length: 8}),
      username: chance.word(),
    });
  });

  describe('GET /api/profiles/:username', () => {
    describe('when the username is not found', () => {
      test('should return a not found error', async () => {
        const {body, status} = await request(app)
          .get('/api/profiles/invalid')
          .set('Accept', 'application/json')
          .send();

        expect({body, status}).toMatchSnapshot();
      });
    });

    describe('when the user is not authenticated', () => {
      test('should return a user profile', async () => {
        const {body, status} = await request(app)
          .get(`/api/profiles/${userOne.get('username')}`)
          .set('Accept', 'application/json')
          .send();

        expect({body, status}).toMatchSnapshot();
      });
    });

    describe('when the user is authenticated', () => {
      test('should return a user profile', async () => {
        const {body, status} = await request(app)
          .get(`/api/profiles/${userTwo.get('username')}`)
          .set('Accept', 'application/json')
          .set('Authorization', `Token ${users.generateJWT(userOne)}`)
          .send();

        expect({body, status}).toMatchSnapshot();
      });
    });
  });
});
