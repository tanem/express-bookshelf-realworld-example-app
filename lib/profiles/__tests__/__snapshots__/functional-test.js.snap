// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profiles GET /api/profiles/:username when the user is authenticated should return a user profile 1`] = `
{
  "body": {
    "profile": {
      "bio": "Contract Administrator",
      "following": false,
      "image": "//www.gravatar.com/avatar/381695debd1a3738c5b4bb61d125ceb9",
      "username": "wochaco",
    },
  },
  "status": 200,
}
`;

exports[`profiles GET /api/profiles/:username when the user is not authenticated should return a user profile 1`] = `
{
  "body": {
    "profile": {
      "bio": "Interpreter",
      "following": false,
      "image": "//www.gravatar.com/avatar/bbd7b116dd01d6f12cb1b279a4625885",
      "username": "ela",
    },
  },
  "status": 200,
}
`;

exports[`profiles GET /api/profiles/:username when the username is not found should return a not found error 1`] = `
{
  "body": {
    "errors": {
      "error": {
        "data": null,
        "isBoom": true,
        "isServer": false,
        "message": "Not Found",
        "output": {
          "headers": {},
          "payload": {
            "error": "Not Found",
            "message": "Not Found",
            "statusCode": 404,
          },
          "statusCode": 404,
        },
      },
      "message": "Not Found",
    },
  },
  "status": 404,
}
`;
